### PR TITLE
Optimize dynamic configuration query services

### LINKED ISSUE #1523: Refactor dynamic configuration services
### What would you like to be enhanced?

The dynamic configuration service is optimized, and a common client interface is added.

### Why is this needed?

In this way, the backend can directly use the dynamic configuration interface of the sermant-agentcore-implement for configuration processing.

### PR DESCRIPTION
**Pull request by hanbingleixue**
**What type of PR is this?**

Enhancement.

**What this PR does / why we need it?**

In this way, the backend can directly use the dynamic configuration interface of the sermant-agentcore-implement for configuration processing..

**Which issue(s) this PR fixes？**

Fixes #1523 

**Does this PR introduce a user-facing change?**

No

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/sermant-io/Sermant/issues) related with this PR before you start working on it.
- [x] Make sure you have squashed your change to one single commit.
- [x] GitHub Actions works fine in this PR.

### CODE REVIEW

File: sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/ConfigClient.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ */
+public interface ConfigClient {
+    /**
+     * Get configuration
+     *
+     * @param key configuration key
+     * @param group configuration group
+     * @return configuration content
+     */
+    String getConfig(String key, String group);
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/ConfigClient.java:36]: I think namespace parm is need

File: sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/ConfigClient.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    /**
+     * Get configurations
+     *
+     * @param key configuration key
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/ConfigClient.java:41]: configuration key for fuzzy search. configuration group for fuzzy search

File: sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/common/Constants.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+package io.sermant.implement.service.dynamicconfig.common;
+
+/**
+ * Constant class
+ *
+ * @author zhp
+ * @since 2024-05-21
+ */
+public class Constants {
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/common/Constants.java]: DynamicConstants？

File: sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/kie/client/kie/KieClient.java [hunk 4]
```diff
@@ -178,11 +178,99 @@ public boolean doDeleteConfig(String keyId) {
...
+    public Map<String, List<String>> getConfigList(String key, String group, String namespace) {
+        String labelCondition = LabelGroupUtils.getLabelCondition(group);
+        final KieRequest cloneRequest = new KieRequest().setRevision(null).setLabelCondition(labelCondition);
+        final KieResponse kieResponse = this.queryConfigurations(cloneRequest);
+        if (kieResponse == null || kieResponse.getData() == null) {
+            return Collections.emptyMap();
+        }
+        Map<String, List<String>> result = new HashMap<>();
+        for (KieConfigEntity entity : kieResponse.getData()) {
+            if (key.equals(entity.getKey()) || entity.getKey().contains(key)) {
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/kie/client/kie/KieClient.java]: remove key.equals(entity.getKey())

File: sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/kie/client/kie/KieClient.java [hunk 5]
```diff
@@ -178,11 +178,99 @@ public boolean doDeleteConfig(String keyId) {
...
+     * Get key_id
+     *
+     * @param key configuration key
+     * @param group configuration group
+     * @return key_id, return null if not exists
+     */
+    public Optional<String> getKeyId(String key, String group) {
+        String labelCondition = LabelGroupUtils.getLabelCondition(group);
+        final KieRequest cloneRequest = new KieRequest().setRevision(null).setLabelCondition(labelCondition);
+        final KieResponse kieResponse = this.queryConfigurations(cloneRequest);
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/kie/client/kie/KieClient.java]: 242-245 extract as a function

File: sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/kie/client/kie/KieClient.java [hunk 6]
```diff
@@ -178,11 +178,99 @@ public boolean doDeleteConfig(String keyId) {
...
+        for (KieConfigEntity entity : kieResponse.getData()) {
+            if (isSameKey(entity, key, labels)) {
+                return Optional.of(entity.getId());
+            }
+        }
+        return Optional.empty();
+    }
+
+    private boolean isSameKey(KieConfigEntity entity, String targetKey, Map<String, String> targetLabels) {
+        if (targetKey != null && targetKey.equals(entity.getKey())) {
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/kie/client/kie/KieClient.java]: Confirm the code logic.

File: sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/nacos/NacosClient.java [hunk 7]
```diff
@@ -0,0 +1,318 @@
...
+     *
+     * @param key configuration key
+     * @param group configuration group
+     * @param timeout Request timeout
+     * @return configuration content
+     * @throws NacosException Configuration query exception
+     */
+    public String getConfig(String key, String group, long timeout) throws NacosException {
+        final String data = this.configService.getConfig(key, group, timeout);
+        return data == null ? "" : data;
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/nacos/NacosClient.java]: Constants.EMPTY_STRING

File: sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/kie/client/kie/KieClient.java [hunk 8]
```diff
@@ -178,11 +178,99 @@ public boolean doDeleteConfig(String keyId) {
...
+            }
+        }
+        return result;
+    }
+
+    @Override
+    public boolean removeConfig(String key, String group) {
+        Optional<String> keyIdOptional = getKeyId(key, group);
+        if (!keyIdOptional.isPresent()) {
+            return false;
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/kie/client/kie/KieClient.java:241]: add a log

File: sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/zookeeper/ZooKeeperBufferedClient.java [hunk 9]
```diff
@@ -107,7 +101,7 @@ public void process(WatchedEvent event) {
      * @param key key for encryption
      */
     public ZooKeeperBufferedClient(String connectString, int sessionTimeout, String userName,
-            String password, String key) {
+                                   String password, String key) {
         String authInfo = userName + ZK_AUTH_SEPARATOR + AesUtil.decrypt(key, password).orElse(null);
         zkClient = newZkClient(connectString, sessionTimeout, new Watcher() {
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/zookeeper/ZooKeeperBufferedClient.java]: new ZooKeeperClient directly

File: sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/zookeeper/ZooKeeperBufferedClient.java [hunk 10]
```diff
@@ -73,15 +67,15 @@ public class ZooKeeperBufferedClient implements Closeable {
...
      * Create a ZooKeeperBufferedClient, initialize the ZK client, and provide an expired reconnection mechanism
      *
      * @param connectString connect string, must be in the following format: {@code host:port[(,host:port)...]}
      * @param sessionTimeout session timeout
      * @throws ZooKeeperInitException In the case of dependent dynamic configuration, if ZK initialization fails then
-     * Sermant needs to be interrupted
+     *                                Sermant needs to be interrupted
      */
     public ZooKeeperBufferedClient(String connectString, int sessionTimeout) {
         zkClient = newZkClient(connectString, sessionTimeout, new Watcher() {
```
Comment (lilai23) [inline sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/zookeeper/ZooKeeperBufferedClient.java]: new ZooKeeperClient

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/utils/LabelGroupUtils.java [hunk 11]
```diff
@@ -43,26 +38,6 @@ public class LabelGroupUtils {
...
-     * Key-value pair length
-     */
-    private static final int KV_LEN = 2;
-
-    /**
-     * Default group
-     */
-    private static final String DEFAULT_GROUP_KEY = "GROUP";
-
     private LabelGroupUtils() {
```
Comment (lilai23) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/utils/LabelGroupUtils.java]: move remained method createLabelGroup to AbstractGroupConfigSubscriber for its usage

File: sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/ConfigClient.java (hunk 1 above)
Comment (hanbingleixue) [reply sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/ConfigClient.java:36]: No namespace is required. One nacosClient corresponds to one namespace.

File: sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/ConfigClient.java (hunk 2 above)
Comment (hanbingleixue) [reply sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/ConfigClient.java:41]: The parameter of the matching mode is added.

File: sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/common/Constants.java (hunk 3 above)
Comment (hanbingleixue) [reply sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/common/Constants.java]: done

File: sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/kie/client/kie/KieClient.java (hunk 4 above)
Comment (hanbingleixue) [reply sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/kie/client/kie/KieClient.java]: done

File: sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/kie/client/kie/KieClient.java (hunk 5 above)
Comment (hanbingleixue) [reply sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/kie/client/kie/KieClient.java]: done

File: sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/kie/client/kie/KieClient.java (hunk 6 above)
Comment (hanbingleixue) [reply sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/kie/client/kie/KieClient.java]: done

File: sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/nacos/NacosClient.java (hunk 7 above)
Comment (hanbingleixue) [reply sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/nacos/NacosClient.java]: done

File: sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/kie/client/kie/KieClient.java (hunk 8 above)
Comment (hanbingleixue) [reply sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/kie/client/kie/KieClient.java:241]: done

File: sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/zookeeper/ZooKeeperBufferedClient.java (hunk 9 above)
Comment (hanbingleixue) [reply sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/zookeeper/ZooKeeperBufferedClient.java]: done

File: sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/zookeeper/ZooKeeperBufferedClient.java (hunk 10 above)
Comment (hanbingleixue) [reply sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/zookeeper/ZooKeeperBufferedClient.java]: done

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/utils/LabelGroupUtils.java (hunk 11 above)
Comment (hanbingleixue) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/utils/LabelGroupUtils.java]: done

**Review by daizhenyu — APPROVED** [review]

**Review by lilai23 — APPROVED** [review]

**Pull request merged**